corehq/apps/locations/fixtures.py

```diff
@@ -53,20 +56,25 @@ def location_fixture_generator(user, version, last_sync=None):
         return []
 
     if toggles.SYNC_ALL_LOCATIONS.enabled(user.domain):
-        location_db = _location_footprint(Location.by_domain(user.domain))
+        unfiltered_locations = SQLLocation.objects.filter(domain=user.domain)
+        locations = [loc for loc in unfiltered_locations if _valid_parent_type(loc)]
     else:
         locations = []
         if user.location:
             # add users location (and ancestors) to fixture
-            locations.append(user.location)
+            locations.append(user.sql_location)
 
             # add all descendants as well
-            locations += user.location.descendants
+            locations += user.sql_location.get_descendants()
 
         if user.project.supports_multiple_locations_per_user:
             # this might add duplicate locations but we filter that out later
-            locations += user.locations
-        location_db = _location_footprint(locations)
+            location_ids = [loc._id for loc in user.locations]
+            locations += SQLLocation.objects.filter(
+                location_id__in=location_ids
+            )
```

Comment:
Actually, no. The other check was pointless. The list gets filtered right before it gets passed to `_location_footprint` and then every parent is checked before being added.